contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql

```diff
@@ -2543,6 +2543,30 @@ GRANT EXECUTE ON FUNCTION sys.has_perms_by_name(
     permission sys.SYSNAME, 
     sub_securable sys.SYSNAME,
     sub_securable_class sys.nvarchar(60)) TO PUBLIC;
+    
+CREATE OR REPLACE PROCEDURE sys.analyze_babelfish_catalogs()
+LANGUAGE plpgsql
+AS $$ 
+DECLARE 
+	babelfish_catalog RECORD;
+	schema_name varchar = 'sys';
+	error_msg text;
+BEGIN
+	FOR babelfish_catalog IN (
+		SELECT relname as name from pg_class t 
+		INNER JOIN pg_namespace n on n.oid = t.relnamespace
+		WHERE t.relkind = 'r' and n.nspname = schema_name
+		)
+	LOOP
+		BEGIN
+			EXECUTE format('ANALYZE %I.%I', schema_name, babelfish_catalog.name);
+		EXCEPTION WHEN OTHERS THEN
+			GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+			RAISE WARNING 'ANALYZE for babelfish catalog %.% failed with error: %s', schema_name, babelfish_catalog.name, error_msg;
```

Comment:
When we have this warning in the upgrade, what's our action item ?

Reply:
Warning will help us to debug any performance issue after upgrade due the failure. I will add the same in Jira also.

Reply:
how will it lead to action on customer or our side? A warning will go to upgrade scripts and might be lost if no one takes a look.

Reply:
yes, warning will not lead to any action but it will be helpful  in case of any performance issue observed after the upgrade. If logs are lost in that case we have to investigate the issue from beginning.

Reply:
Let us track this item via a Jira. Talk to Sharu about this. We should consider how we track successful MVUs and corresponding logs in long term to resolve issues down the line.